lib/davinci_dtr_test_kit/client_groups/dinner_static/dtr_full_ehr_dinner_static_questionnaire_response_correctness_test.rb

```diff
@@ -23,7 +23,8 @@ class DTRFullEHRDinnerStaticQuestionnaireResponseCorrectnessTest < Inferno::Test
       # questionnaire = Fixtures.find_questionnaire('DinnerOrderStatic')
       qr = FHIR.from_contents(questionnaire_response)
       check_origin_sources(questionnaire.item, qr.item, expected_overrides: ['PBD.2'])
-      check_answer_presence(qr.item, link_ids: ['PBD.1', 'PBD.2', 'LOC.1'])
+      required_link_ids = extract_required_link_ids(questionnaire.item)
+      check_answer_presence(qr.item, required_link_ids)
```

Comment:
In this test we are checking the presence of answers based on the workflow rather than what is required in the questionnaire. The workflow technically only asks the user to confirm that the questionnaire is pre-populated (PBD.1, PBD.2, LOC.1), and that rendering is accurate, but it doesn't ask the user to attest that they've completed the questionnaire.

We do say in the description for this group:
> The tester will complete the questionnaire such that a QuestionnaireResponse is stored back into Inferno's EHR endpoint

But that feels a little soft/subtle to me to then fail them if they haven't completed all required fields. Presumably you could save the QuestionnaireResponse to the EHR without completing it.

Although now that I'm thinking about it, I'm wondering if we should actually be checking for presence of all required fields. But @karlnaden approved this as is, so I want to check with Karl on the intent for this test. Should this be all required fields? Should we modify the description or attestations at all?

Reply:
I agree that we should be checking that all required fields have been populated and attestations should be clear that the quesitonnaire should be completed and stored back to the EHR.